MapboxNavigation/CarPlayManager.swift

```diff
@@ -406,16 +406,12 @@ extension CarPlayManager: CPListTemplateDelegate {
         }
         
         if let error = error {
-            let okTitle = NSLocalizedString("CARPLAY_OK", bundle: .mapboxNavigation, value: "OK", comment: "CPNavigationAlert OK button title")
-            let okAction = CPAlertAction(title: okTitle, style: .default) { _ in
-                interfaceController.popToRootTemplate(animated: true)
+            guard let delegate = delegate,
+                  let alert = delegate.carplayManager?(self, didFailToFetchRouteBetweenWaypoints: waypoints, options: routeOptions, error: error) else {
+                    return
             }
-            let alert = CPNavigationAlert(titleVariants: [error.localizedDescription],
-                                          subtitleVariants: [error.localizedFailureReason ?? ""],
-                                          imageSet: nil,
-                                          primaryAction: okAction,
-                                          secondaryAction: nil,
-                                          duration: 0)
+
+            interfaceController.popToRootTemplate(animated: true)
```

Comment:
I'm not so sure of this one - seems like a POLS violation, and the user may lose context on their current activity.

I'd like to try this out on a device when ready & we can figure out if there are any alternatives in order to successfully display an alert.

Reply:
posterity: let's try a `CPAlertTemplate` here so we don't have to blow the stack away.

Reply:
Update: looks like a `CPAlertTemplate` is not possible.

```swift
2019-02-18 15:37:34.803372-0700 Example-CarPlay[23447:1959284] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Unsupported object <CPAlertTemplate: 0x600001272580> <identifier: 6638156E-21B0-435F-BD42-DB2B40173057, userInfo: (null)> passed to pushTemplate:animated:. Allowed classes: {(
    CPSearchTemplate,
    CPMapTemplate,
    CPListTemplate,
    CPGridTemplate
)}
```